docs/examples/starknet-counter.mdx

```diff
@@ -0,0 +1,485 @@
+  nodeUri: Uri.parse(
+      'http://localhost:5050'),
+);
+
+Future<int> getCurrentCount() async {
+  final result = await provider.call(
+    request: FunctionCall(
+        contractAddress: Felt.fromHexString(contractAddress),
+        entryPointSelector: getSelectorByName("get_current_count"),
+        calldata: []),
+    blockId: BlockId.latest,
+  );
+  return result.when(
+    result: (result) => result[0].toInt(),
+    error: (error) => throw Exception("Failed to get counter value"),
+  );
+}
+
+Future<String> increaseCounter() async {
+  print('print increment');
+  final response = await signeraccount.execute(functionCalls: [
+    FunctionCall(
+      contractAddress: Felt.fromHexString(contractAddress),
+      entryPointSelector: getSelectorByName("increment"),
+      calldata: [],
+    ),
+  ]);
+
+  final txHash = response.when(
+    result: (result) => result.transaction_hash,
+    error: (err) => throw Exception("Failed to execute"),
+  );
+
+  print('printing increment TX : $txHash');
+  return txHash;
+  // return waitForAcceptance(transactionHash: txHash, provider: provider);
+}
+
+Future<String> increaseCounterBy(String number) async {
+  print('print increment by ');
+  final response = await signeraccount.execute(functionCalls: [
+    FunctionCall(
+      contractAddress: Felt.fromHexString(contractAddress),
+      entryPointSelector: getSelectorByName("increase_count_by"),
+      calldata: [Felt.fromIntString(number)],
+    ),
+  ]);
+
+  final txHash = response.when(
+    result: (result) => result.transaction_hash,
+    error: (err) => throw Exception("Failed to execute"),
+  );
+
+  print('printing incrementby amount TX : $txHash');
+  return txHash;
+  // return waitForAcceptance(transactionHash: txHash, provider: provider);
+}
+
+Future<String> decreaseCounter() async {
+  print('decrementing.....');
+  final response = await signeraccount.execute(functionCalls: [
+    FunctionCall(
+      contractAddress: Felt.fromHexString(contractAddress),
+      entryPointSelector: getSelectorByName("decrement"),
+      calldata: [],
+    ),
+  ]);
+
+  final txHash = response.when(
+    result: (result) => result.transaction_hash,
+    error: (err) => throw Exception("Failed to execute"),
+  );
+  print('printing decrement TX : $txHash');
+  return txHash;
+  // return waitForAcceptance(transactionHash: txHash, provider: provider);
+}
+```
+
+6. Create `lib/ui/counter.dart` with the following content:
+```dart
+import '../services/counter_service.dart';
+import 'package:flutter/material.dart';
+
+class CounterPage extends StatefulWidget {
+  const CounterPage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<CounterPage> createState() => _CounterPageState();
+}
+
+class _CounterPageState extends State<CounterPage> {
+  int counter = 0;
+  TextEditingController amount = TextEditingController();
+  _increaseCount() async {
+    await increaseCounter();
+    await _getCounter();
```

Comment:
Please note that In `increaseCounter` call, there is no wait for transaction acceptance, which means that `getCurrentCount` could retrieve previous value.
 Contract state will only been updated once transaction have been accepted.

Reply:
waitForAcceptance added in counter service.